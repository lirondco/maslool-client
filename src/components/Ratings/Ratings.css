.trail_rating {
  margin-bottom: 2%;
  padding: 1%;
  border: #524c57 solid;
  border-radius: 10px;
  box-shadow: 5px 5px 2.5px grey;
}

.rating_date {
  font-style: italic;
}

.rating_content {
  padding: 1% 3%;
}
