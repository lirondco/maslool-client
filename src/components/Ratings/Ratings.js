import React, { Component } from 'react'
import StarRatings from 'react-star-ratings'
import { NiceDate } from '../Utils/Utils'
import './Ratings.css'

export default class Ratings extends Component {


    render() {
        const rating = this.props.rating

        return (
            <div className='trail_rating'>
                <h4 className='rating_username'>From: {rating.user.username} {(rating.user.banned) ? '[banned]' : ''}{(rating.user.admin) ? '[moderator]' : ''}</h4>
                <div className='rating_content'>
                    <StarRatings
                        rating={rating.rating}
                        starDimension='20px'
                        starSpacing='5px'
                        starRatedColor='red'
                        starEmptyColor='grey'
                    />
                </div>
                <p className='rating_date'>modified: <NiceDate date={rating.date_modified} /></p>
            </div>
        )
    }
}