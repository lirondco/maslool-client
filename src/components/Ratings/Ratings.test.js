import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Ratings from "./Ratings";

describe("Ratings component", () => {
  const props = {
    id: 7,
    rating: 3,
    date_modified: "2021-03-11T06:47:06.477Z",
    user: {
        id: 1,
        username: "test_user",
        join_date: "2021-03-11T06:47:06.477Z",
        admin: false,
        banner: false
    }
  };

  it("renders the rating given props", () => {
    const wrapper = shallow(<Ratings rating={props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  
});
