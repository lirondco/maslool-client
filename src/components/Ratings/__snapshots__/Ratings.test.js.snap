// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ratings component renders the rating given props 1`] = `
<div
  className="trail_rating"
>
  <h4
    className="rating_username"
  >
    From: 
    test_user
     
  </h4>
  <div
    className="rating_content"
  >
    <StarRatings
      changeRating={null}
      gradientPathName=""
      ignoreInlineStyles={false}
      numberOfStars={5}
      rating={3}
      starDimension="20px"
      starEmptyColor="grey"
      starHoverColor="rgb(230, 67, 47)"
      starRatedColor="red"
      starSpacing="5px"
      svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
      svgIconViewBox="0 0 51 48"
      typeOfWidget="Star"
    />
  </div>
  <p
    className="rating_date"
  >
    modified: 
    <NiceDate
      date="2021-03-11T06:47:06.477Z"
    />
  </p>
</div>
`;
